File: project2v2/receiver.py
#!/usr/bin/env python3

import sys 

from rdt import RDTSocket

def main():
    if len(sys.argv) < 4:
        print("Usage: receiver <port> <window_size> <filename>")
        exit(1)

    port = int(sys.argv[1])
    window_size = int(sys.argv[2])
    filename = sys.argv[3]

    socket = RDTSocket(window_size)
    socket.bind("localhost", port)
    socket.accept()

    data = socket.receive()

    with open(filename, "w") as f:
        f.write(data.decode())
 
if __name__ == "__main__":
    main()

File: project2/src/receiver.cpp
#include <cstdint>
#include <iostream>
#include <fstream>

#include "util.h"
#include "rdt.h"


int main(int argc, char** argv) {
    argc--;
    argv++;

    if(argc < 2) {
        std::cout << "Invalid Usage." << std::endl;
        std::cout << "Usage: receiver <recv port> <window size>" << std::endl;
        return 1;
    }

    const auto port = parse_int<uint16_t>(argv[0]);
    const auto window_size = parse_int<uint32_t>(argv[1]);

    
    RDTSocket socket(window_size);

    socket.accept();
    
    auto [data, len] = socket.recv();

    std::ofstream file("download.txt");
    file.write((char*) data, len);
    file.close();
    return 0;
}
